frontend/talentsearch/src/js/components/workLocationPreferenceForm/WorkLocationPreferenceForm.tsx

```diff
@@ -0,0 +1,205 @@
+import { commonMessages, errorMessages } from "@common/messages";
+import React from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { useIntl } from "react-intl";
+import { toast } from "react-toastify";
+import { useApplicantProfileRoutes } from "../../applicantProfileRoutes";
+import {
+  CreateUserInput,
+  CreateWorkLocationPreferenceMutation,
+  useCreateWorkLocationPreferenceMutation,
+  WorkRegion,
+  User,
+  UpdateUserAsUserInput,
+  useWorkLocationPreferenceQuery,
+} from "../../api/generated";
+import ProfileFormFooter from "../applicantProfile/ProfileFormFooter";
+import ProfileFormWrapper from "../applicantProfile/ProfileFormWrapper";
+
+export type FormValues = Pick<
+  CreateUserInput,
+  "locationPreferences" | "locationExemptions"
+>;
+interface WorkLocationPreferenceFormProps {
+  initialData: User;
+  handleWorkLocationPreference: (
+    id: string,
+    data: UpdateUserAsUserInput,
+  ) => Promise<CreateWorkLocationPreferenceMutation["updateUserAsUser"]>;
+}
+
+export const WorkLocationPreferenceForm: React.FC<
+  WorkLocationPreferenceFormProps
+> = ({ initialData, handleWorkLocationPreference }) => {
+  const intl = useIntl();
+
+  function bold(msg: string) {
+    return <span data-h2-font-weight="b(700)">{msg}</span>;
+  }
+
+  const dataToFormValues = (data: User): FormValues => ({
+    ...data,
+    locationPreferences: data.locationPreferences,
+    locationExemptions: data.locationExemptions,
+  });
+  const formValuesToSubmitData = (
+    values: FormValues,
+  ): UpdateUserAsUserInput => ({
+    locationPreferences: values.locationPreferences,
+    locationExemptions: values.locationExemptions,
+  });
+  const methods = useForm<FormValues>({
+    defaultValues: dataToFormValues(initialData),
+  });
+  const { handleSubmit } = methods;
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    await handleWorkLocationPreference(
+      initialData.id,
+      formValuesToSubmitData(data),
+    );
+  };
+
+  return (
+    <ProfileFormWrapper
+      description={intl.formatMessage({
+        defaultMessage:
+          "Indicate all locations where you are willing to work, including your current location (if you are interested in working there).",
+        description:
+          "Description text for Profile Form wrapper  in Work Location Preferences Form",
+      })}
+      title={intl.formatMessage({
+        defaultMessage: "Work location",
+        description:
+          "Title for Profile Form wrapper  in Work Location Preferences Form",
+      })}
+      crumbs={[
+        {
+          title: intl.formatMessage({
+            defaultMessage: "Work Location Preference",
+            description:
+              "Display Text for the current page in Work Location Preference Form Page",
+          }),
+        },
+      ]}
+    >
+      <FormProvider {...methods}>
+        <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
Can you extract these elements, and ProfileFormFooter, out of this component and put them in the wrapper component instead (WorkLocationPreferenceApi)? There's nothing wrong with the way you have it here, in fact its how I originally recommended, but I was convinced by #2323 that putting these form elements in the wrapper component is a better way of organizing this. It means this component can be more focused on presentation, and the wrapper more focused on function (including defining all the data transformation and submit functions).

Reply:
M... I am waiting to merge that [PR](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2323) to merge to main. I am wondering it may break the code. Now its working fine I am so happy I could display and update the database data. Since you are recommending I will try to do that. Thank you!

Reply:
@tristan-orourke I was trying to work through this refactor with Vany and I think there are some difficulties with this approach.  The "inner" component is what can be cleanly staged for storybook and jest tests and If you move stuff out of it then it makes it tricky to get a good story with a mocked handler or run a good test.  For example, in the #2323 storybook he's had to recreate a substitute wrapper for the story to make it look the same.  If he staged just the GovernmentInfoForm then he couldn't get the handler mocked for the story.  I'm not convinced this is the better approach.

Reply:
@petertgiles directed me to this discussion in #2413 and when working on that, I initially used the inner component for display only but ran into the issue Peter mentioned in that I was essentially recreating it to write proper tests and stories so I moved it back.

I made the actual form component smart enough that I could run tests against it but it simply has the task for normalizing and bubbling data up to its parent component to handle the actual loading/mutating of data.

Reply:
I see your points, I might have been too hasty with the other approach 😅 . Let's go with a version based on @esizer's PR, #2413.
I can see us wanting to reuse the layout of the form inputs in the future within a larger form, which is what I was thinking about with #2323, but that could be extracted pretty easily from this version if necessary.

Reply:
Peter linked me here, so not entirely sure what the context is. But my component explicitly followed a specific layout for the sole reason that you cannot use React Hooks without being inside a ReactForm, therefore the Form logic was needed to be in the outside component so Hooks could exist in the inner component. 
If I did not need Hooks then layout options increases immensely.